akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -30,6 +30,14 @@ sealed abstract case class Uri(scheme: String, authority: Authority, path: Path,
   def isEmpty: Boolean
 
   /**
+   * Parses the given String into a Query instance.
+   */
+  def query(charset: Charset = UTF8, mode: Uri.ParsingMode = Uri.ParsingMode.Relaxed): Query = queryString match {
```

Comment:
Btw. is there a difference between `queryString = Some("")` and `queryString = None`? Otherwise, you could get rid of the option wrapper. (I guess it's whether there's a `'?'` sign or not. We need to decide if we want to keep this distinction.)